fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt

```diff
@@ -0,0 +1,248 @@
+
+        onboardingStore = StoreProvider.get(this) {
+            OnboardingStore(
+                initialState = OnboardingFragmentState(
+                    onboardingState = onboardingAccountObserver.getOnboardingState(),
+                ),
+            )
+        }
+
+        _interactor = DefaultOnboardingInteractor(
+            controller = DefaultOnboardingController(
+                activity = activity,
+                navController = findNavController(),
+                hideOnboarding = ::hideOnboarding,
+            ),
+            privateBrowsingController = DefaultPrivateBrowsingController(
+                activity = activity,
+                appStore = requireComponents.appStore,
+                navController = findNavController(),
+            ),
+            searchSelectorController = DefaultSearchSelectorController(
+                activity = activity,
+                navController = findNavController(),
+            ),
+            toolbarController = DefaultToolbarController(
+                activity = activity,
+                store = requireComponents.core.store,
+                navController = findNavController(),
+            ),
+        )
+
+        toolbarView = ToolbarView(
+            binding = binding,
+            context = requireContext(),
+            interactor = interactor,
+        )
+
+        onboardingView = OnboardingView(
+            containerView = binding.sessionControlRecyclerView,
+            interactor = interactor,
+        )
+
+        activity.themeManager.applyStatusBarTheme(activity)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        consumeFrom(onboardingStore) { state ->
+            onboardingView?.update(state.onboardingState, requireComponents.fenixOnboarding.config)
+        }
+
+        homeMenuView = HomeMenuView(
+            view = view,
+            context = view.context,
+            lifecycleOwner = viewLifecycleOwner,
+            homeActivity = activity as HomeActivity,
+            navController = findNavController(),
+            menuButton = WeakReference(binding.menuButton),
+            hideOnboardingIfNeeded = interactor::onStartBrowsingClicked,
+        ).also { it.build() }
+
+        tabCounterView = TabCounterView(
+            context = requireContext(),
+            browsingModeManager = browsingModeManager,
+            navController = findNavController(),
+            tabCounter = binding.tabButton,
+        )
+
+        toolbarView?.build()
+
+        PrivateBrowsingButtonView(
+            button = binding.privateBrowsingButton,
+            browsingModeManager = browsingModeManager,
+            onClick = { mode ->
+                interactor.onPrivateModeButtonClicked(mode, userHasBeenOnboarded = false)
+            },
+        )
+
+        consumeFrom(store) {
+            tabCounterView?.update(it)
+        }
```

Comment:
nit: maybe block this next to the `tabCounterView` initialization?

Reply:
Done